lms/djangoapps/courseware/access.py

```diff
@@ -136,8 +136,9 @@ def has_access(user, action, obj, course_key=None):
     if not user:
         user = AnonymousUser()
 
-    if in_preview_mode():
-        if not bool(has_staff_access_to_preview_mode(user=user, obj=obj, course_key=course_key)):
+    # Preview mode is only accessible by staff.
+    if in_preview_mode() and course_key:
```

Comment:
OK, from this comment, I thought that is_preview_mode would never return True if course_key was None: "It will not be possible to be in preview mode without course_key".